spec/models/guest_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe Guest, type: :model do
+  describe 'relationships' do
+    it { should have_many :guest_rooms }
+    it { should have_many(:rooms).through(:guest_rooms) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of :name }
+  end
+
+  describe 'class methods' do
+    before do 
+      Guest.create!(name: "Letti")
+      Guest.create!(name: "Calum")
+      Guest.create!(name: "Nika")
+    end
+
+    it "::total_count" do
+      expect(Guest.total_count).to eq(3)
+
+      Guest.create!(name: "Melony")
+
+      expect(Guest.total_count).to eq(4)
```

Comment:
So, this is a tricky method to test... You implemented`total_count` as a Class method on Guest. This is fine, but when we do this, we have 2 different ways we can call the method --- on a collection of ALL the guests, or a collection of SOME of the guests based on an associated Room. The 2nd way is what the user story asks for, and it's how you have it implemented in your View and Feature tests, which is great.

However, we also need to be sure we're testing for that in the model test - the model tests are unit/atomic tests that ensure the functionality we designed is working the way it's intended. This test is looking at calling on ALL the guests, we just need to have additional tests to call on specific associations of Rooms, which is how you're using the method in the view/elsewhere.

So, if this included something like: 
```
room1 = Room.create(...)
room2 = Room.create(...)

GuestRoom.create(guest: mel, room: room1)
GuestRoom.create(guest: calum, room: room1)

expect(room1.guests.count_guests).to eq(2)
```
then that would be a complete test. As it stands now though, I would say this model is not completely tested.